### PR TITLE
bug(parse): Capture `EndIf` token when used alongside `else`

### LINKED ISSUE #392: End If token undefined when dealing with else or elseif
When parsing conditional statements, the `end if` token is sometimes undefined. This seems to happen anytime an `else` keyword is present.

### PR DESCRIPTION
**Pull request by TwitchBronBron**
Fixes #392

### CODE REVIEW

File: test/componentprocessor/componentprocessor.test.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ describe.only("component parsing support", () => {
 
                 const expectedPrefix = "pkg:/test/componentprocessor/resources/scripts/";
                 let expectedScripts = ["extendedComponent.brs", "utility.brs"].map(scriptName =>
-                    path.join(expectedPrefix, scriptName)
+                    path.join(expectedPrefix, scriptName).replace(/\\/g, "/")
```
Comment (TwitchBronBron) [inline test/componentprocessor/componentprocessor.test.js]: This test was failing on my Windows computer. It's unrelated to this PR, but was a one-liner, figured I would include it. I can pull it out into its own PR if you prefer.

Comment (sjbarag) [reply test/componentprocessor/componentprocessor.test.js:101]: Strange!  A separate PR would be lovely — I suspect there's a few cases where we use `path.join()` in an attempt to be portable when we actually want `path.posix.join()` https://nodejs.org/api/path.html#path_windows_vs_posix , and it'll be easier to review those in isolation 😄

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,4 @@ lib/
 types/
 node_modules/
 .vscode/
+coverage
```
Comment (sjbarag) [inline .gitignore:6]: What produces a `coverage` file (or directory) that required this?

File: src/parser/Parser.ts [hunk 3]
```diff
@@ -873,7 +873,7 @@ export class Parser {
 
                     match(Lexeme.Newline);
                     elseBranch = block(Lexeme.EndIf);
-                    let endIfToken = advance(); // skip past "end if"
+                    endIfToken = advance();
```
Comment (sjbarag) [inline src/parser/Parser.ts:876]: Dang, variable shadowing!  Great catch ❤️

File: .gitignore (hunk 2 above)
Comment (TwitchBronBron) [reply .gitignore]: Running the coverage report using jest (`npx jest --coverage`). I undid the change.

File: test/componentprocessor/componentprocessor.test.js (hunk 1 above)
Comment (TwitchBronBron) [reply test/componentprocessor/componentprocessor.test.js]: k. I'll submit a separate PR with just this fix in it.

**Comment by TwitchBronBron** [comment]
Changes made.

**Pull request merged**